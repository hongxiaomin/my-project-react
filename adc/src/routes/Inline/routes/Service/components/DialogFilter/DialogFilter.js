import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { injectIntl } from 'react-intl'
import { connect } from 'react-redux'
import { Button, Dialog } from '@blueprintjs/core'
import { Select, Input } from 'antd'
import { isEmpty } from 'lodash'
import classes from './DialogFilter.scss'
import { showMessage } from 'Utils'
import { sendCommand } from '../../modules/Service'

const Option = Select.Option

export class DialogFilter extends React.Component {

  constructor(props) {
    super(props)

    this.state = {
      modelId: '', // this.props.info.modelId,
      serviceName: '', //this.props.info.serviceName,
    }
  }

  componentWillReceiveProps(nextProps) {
    // this.setState({
    //   modelId: nextProps.info.modelId,
    //   serviceName: nextProps.info.serviceName,
    // })
  }

  handleSelectChange = (value) => {
    this.setState({
      modelId: value,
    })
  }

  handleInputChanage = (e) => {
    this.setState({
      serviceName: e.target.value,
    })
  }

  handleSubmit = () => {
    const { info } = this.props
    const { modelId, serviceName } = this.state
    // 对数据进行校验
    if ((serviceName === '' && info.serviceName === '') || (!modelId && !info.modelId)) {
      showMessage({ text: 'serviceName or model is empty', type: 'warning' })
      return
    }

    const obj = {
      serviceId: info.serviceId,
      serviceName: serviceName || info.serviceName,
    }

    if (this.props.info.status === 'none' && (modelId && info.modelId !== modelId)) {
      obj.command = 'init'
      obj.modelId = modelId
    }

    // 对数据进行验证
    this.props.sendCommand(obj, () => {
      this.props.toggleDialog()
    })
  }

  render() {
    const { info, intl } = this.props
    const modelList = this.props.modelList.filter(item => item.isPublished === true)
    return (
      <div className={classes['DialogFilter-container']}>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={classes.headerTitle}>
              {this.props.DialogTitle}
            </h6>
          </div>
          <div className={classes.dialogBody}>
            <div className={classes.filter}>
              <WrapperFormattedMessage id='labelingTool.imageList.name' />
              <div>
                <Input
                  type='text'
                  defaultValue={info.serviceName}
                  onChange={this.handleInputChanage}
                  placeholder={intl.formatMessage({ id: 'serviceName' })}
                />
              </div>
            </div>
            <div className={classes.filter}>
              <WrapperFormattedMessage id='serviceManagement.dialogFilter.model' />
              <div>
                {
                  (!isEmpty(modelList)) ? (
                    <Select
                      defaultValue={info.modelId && info.modelId.toString()}
                      disabled={this.props.info.status !== 'none'}
                      className={classes.select}
                      placeholder={intl.formatMessage({ id: 'pleaseSelect' })}
                      onChange={this.handleSelectChange}
                    >
                      {modelList.map(item => (<Option
                        key={item.id}
                        value={item.id.toString()}
                      >{item.name}</Option>))}
                    </Select>
                  ) : null
                }
              </div>
            </div>
            <div className={classes.btnGroup}>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                <WrapperFormattedMessage id='cancel' />
              </Button>
              <Button
                className={`pt-button pt-intent-primary ${classes.btnEdit}`}
                onClick={this.handleSubmit}
              >
                <WrapperFormattedMessage id='Save' />
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}

DialogFilter.propTypes = {
  intl: PropTypes.object.isRequired,
  info: PropTypes.object.isRequired,
  isOpen: PropTypes.bool.isRequired,
  modelList: PropTypes.array.isRequired,
  DialogTitle: PropTypes.string.isRequired,
  toggleDialog: PropTypes.func.isRequired,
  sendCommand: PropTypes.func.isRequired,
}

export default connect(({ service, intl: { locale } }) => ({
  locale,
  modelList: service.modelList,
}), {
  sendCommand,
})(injectIntl(DialogFilter))
