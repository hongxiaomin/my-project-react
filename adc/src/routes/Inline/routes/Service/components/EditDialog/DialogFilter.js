import React from 'react'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { Select, Input } from 'antd'
import classes from './EditDialog.scss'

const Option = Select.Option
type Props = {
  editOpen: Boolean,
  DialogID: Number,
  DialogTitle:String,
  dataDetail: Object,
  modelList: Array,
  toggleDialog: Function,
  handleSubmit: Function,
  that: Object,
  // dispatch: Function,
};
export class DialogFilter extends React.Component {
  constructor(props) {
    super(props)
    this.state = {

    }
    this.handleChange = this.handleChange.bind(this)
  }

  handleChange(type) {
    return (value) => {
      this.props.that.setState({
        [type]: value,
      })
      // this.props.dispatch({
      //   type: CHANGE_STATUS,
      //   payload: true,
      // })
    }
  }

  props: Props;

  render() {
    return (
      <div>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.editOpen}
            onClose={this.props.toggleDialog}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={this.props.DialogID === '' ? classes.onlyHeaderTitle : classes.headerTitle}>
                {this.props.DialogTitle}
              </h6>
            </div>
            <div className={classes.dialogBody}>
              <div className={classes.filter}>
                Name
                <Input
                  readOnly='true'
                  defaultValue={this.props.dataDetail.serviceName}
                />
              </div>
              {/* <div className={classes.filter}>
                Device ID
                <Input
                  readOnly='true'
                  defaultValue={this.props.dataDetail.deviceId}
                />
              </div> */}
              <div className={classes.filter}>
                 Model
                 <Select
                   name='model'
                   style={{ width: 200 }}
                   placeholder="Please select"
                   onChange={this.handleChange('model')}
                 >
                   {
                     !isEmpty(this.props.modelList) ?
                     this.props.modelList.map(item => (<Option
                       key={item.id}
                     >{item.path}</Option>)) :
                     [].map(item => (<Option
                       key={item.id}
                     >{item.path}</Option>))
                   }
                 </Select>
              </div>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                 Cancel
                </Button>
                <Button
                  className={`pt-button pt-intent-primary ${classes.btnEdit}`}
                  onClick={this.props.handleSubmit}
                >
                  Save
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}
export default DialogFilter
