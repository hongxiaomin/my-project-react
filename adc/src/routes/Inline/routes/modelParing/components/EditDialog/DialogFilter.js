import React from 'react'
import PropTypes from 'prop-types'
import WrapperFormattedMessage from 'components/WrapperFormattedMessage'
import { isEmpty } from 'lodash'
import { Button, Dialog } from '@blueprintjs/core'
import { connect } from 'react-redux'
import { Select, Input, Checkbox } from 'antd'
import classes from './EditDialog.scss'
import { CHANGE_STATUS, getModelDefect } from '../../modules/ModelParing'

const Option = Select.Option
export class DialogFilter extends React.Component {
  constructor(props) {
    super(props)
    this.state = {

    }
    this.handleChange = this.handleChange.bind(this)
    this.checkboxDefault = this.checkboxDefault.bind(this)
  }
  // 是否设置为试运行
  checkboxDefault(e) {
    this.props.that.setState({
      isTrialRun: e.target.checked,
    })
  }

  handleChange(type) {
    return (value) => {
      this.props.that.setState({
        [type]: value,
      }, () => {
        this.props.dispatch({
          type: CHANGE_STATUS,
          payload: true,
        })
        this.props.dispatch(getModelDefect(value))
      })
    }
  }

  render() {
    const modelList = this.props.modelList.filter(item => item.isPublished === true)
    let defectCode = []
    if (this.props.status && this.props.that.state.model !== '' && !isEmpty(this.props.modelDefect)) {
      this.props.modelDefect.map(item => (defectCode.push(item.defectCode)))
    }
    defectCode = defectCode.join('/')
    return (
      <div>
        {
          !isEmpty(this.props.dataDetail) &&
          <Dialog
            isOpen={this.props.editOpen}
            onClose={this.props.toggleEdit}
            className={classes.dialog}
          >
            <div className={classes.header}>
              <h6 className={this.props.DialogID === '' ? classes.onlyHeaderTitle : classes.headerTitle}>
                {this.props.DialogTitle}
              </h6>
            </div>
            <div className={classes.dialogBody}>
              <div className={classes.filter}>
                <WrapperFormattedMessage id='processId' />
                <Input
                  readOnly='true'
                  defaultValue={this.props.dataDetail.original.processName}
                />
              </div>
              <div className={classes.filter}>
                <WrapperFormattedMessage id='productId' />
                <Input
                  readOnly='true'
                  defaultValue={this.props.dataDetail.original.productName}
                />
              </div>
              <div className={classes.filter}>
                <WrapperFormattedMessage id='modelName' />
                <Select
                  showSearch
                  name='model'
                  style={{ width: 200 }}
                  defaultValue={this.props.dataDetail.original.modelName}
                  value={this.props.that.state.model}
                  onChange={this.handleChange('model')}
                >
                  {
                    !isEmpty(modelList) ?
                     modelList.map(item => (<Option
                       key={item.id.toString()}
                       value={item.id.toString()}
                     >{item.name}</Option>)) :
                     [].map(item => (<Option
                       key={item.id.toString()}
                       value={item.id.toString()}
                     >{item.name}</Option>))
                  }
                </Select>
              </div>
              {
                this.props.status ?
                  <div className={classes.filterDefect}>
                    Defect Code
                    <div style={{ width: '200px',
                      wordWrap: 'break-word',
                      backgroundColor: '#eee',
                      margin: '10px',
                      borderRadius: '5px',
                      padding: '10px' }}
                    >
                      {defectCode}
                    </div>
                  </div> :
                  <div />
              }
              <div className={classes.filter}>
                <WrapperFormattedMessage id='modelPairing.dialogFilter.trialRun' />
                <Checkbox
                  style={{ width: '200px', marginLeft: '50px' }}
                  defaultChecked={this.props.dataDetail.original.isTrialRun}
                  onChange={this.checkboxDefault}
                ><WrapperFormattedMessage id='modelPairing.dialogFilter.setAsTrialRun' /></Checkbox>
              </div>
              <div className={classes.btnGroup}>
                <Button className={classes.btnCancel} onClick={this.props.toggleEdit}>
                  <WrapperFormattedMessage id='cancel' />
                </Button>
                <Button
                  className={`pt-button pt-intent-primary ${classes.btnEdit}`}
                  onClick={this.props.handleSubmit}
                >
                  <WrapperFormattedMessage id='Save' />
                </Button>
              </div>
            </div>
          </Dialog>
        }
      </div>
    )
  }
}

DialogFilter.propTypes = {
  editOpen: PropTypes.bool.isRequired,
  DialogID: PropTypes.number.isRequired,
  DialogTitle: PropTypes.string.isRequired,
  dataDetail: PropTypes.object.isRequired,
  modelList: PropTypes.array.isRequired,
  status: PropTypes.bool.isRequired,
  toggleEdit: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  that: PropTypes.object.isRequired,
  modelDefect: PropTypes.array.isRequired,
  dispatch: PropTypes.func.isRequired,
}

export default connect(
  ({ auth, modelParing, intl: { locale } }) => ({
    locale,
    token: auth.token,
    modelDefect: modelParing.modelDefect,
  }),
)(DialogFilter)
