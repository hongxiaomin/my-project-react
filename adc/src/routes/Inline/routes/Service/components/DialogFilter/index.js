import DialogFilter from './DialogFilter'

export default DialogFilter
