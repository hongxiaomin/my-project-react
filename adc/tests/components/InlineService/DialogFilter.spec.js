import React from 'react'
import { DialogFilter } from 'routes/Inline/routes/Service/components/DialogFilter/DialogFilter'

describe('(Component) DialogFilter', () => {
  const info = {}
  const isOpen = false
  
  const DialogTitle = 'dialog'
  const sendCommand = jest.fn()
  const toggleDialog = jest.fn()
  it('render normally', () => {
    const modelList = []
    const wrapper = shallow(
      <DialogFilter
        info={info}
        isOpen={isOpen}
        toggleDialog={toggleDialog}
        modelList={modelList}
        DialogTitle={DialogTitle}
        sendCommand={sendCommand}
      />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('render normally modellist is not null ', () => {
    const modelList = [{ id: '1', name: '1', isPublished: true }]
    const wrapper = shallow(
      <DialogFilter
        info={info}
        isOpen={isOpen}
        toggleDialog={toggleDialog}
        modelList={modelList}
        DialogTitle={DialogTitle}
        sendCommand={sendCommand}
      />,
    )
    wrapper.setState({ serviceName: '' })
    wrapper.find('Input').simulate('change', { target: { value: 'aa' } })
    expect(wrapper.state('serviceName')).toEqual('aa')
    wrapper.find('Button').at(1).simulate('click')
    // expect(wrapper.state('serviceName')).toEqual('aa')
    expect(wrapper).toMatchSnapshot()
  })
})
