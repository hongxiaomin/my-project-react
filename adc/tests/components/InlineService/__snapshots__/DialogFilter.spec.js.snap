// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) DialogFilter render normally 1`] = `
<div
  className="DialogFilter-container"
>
  <Blueprint.Dialog
    className="dialog"
    isOpen={false}
    onClose={[Function]}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        dialog
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <div
        className="filter"
      >
        Name
        <div>
          <Input
            disabled={false}
            onChange={[Function]}
            placeholder="service name"
            prefixCls="ant-input"
            type="text"
          />
        </div>
      </div>
      <div
        className="filter"
      >
        Model
        <div />
      </div>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
          onClick={[Function]}
        >
          Cancel
        </Blueprint.Button>
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Save
        </Blueprint.Button>
      </div>
    </div>
  </Blueprint.Dialog>
</div>
`;

exports[`(Component) DialogFilter render normally modellist is not null  1`] = `
<div
  className="DialogFilter-container"
>
  <Blueprint.Dialog
    className="dialog"
    isOpen={false}
    onClose={[Function]}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        dialog
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <div
        className="filter"
      >
        Name
        <div>
          <Input
            disabled={false}
            onChange={[Function]}
            placeholder="service name"
            prefixCls="ant-input"
            type="text"
            value="aa"
          />
        </div>
      </div>
      <div
        className="filter"
      >
        Model
        <div>
          <Select
            choiceTransitionName="zoom"
            className="select"
            onChange={[Function]}
            placeholder="Please select"
            prefixCls="ant-select"
            showSearch={false}
            transitionName="slide-up"
          >
            <Option
              value="1"
            >
              1
            </Option>
          </Select>
        </div>
      </div>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
          onClick={[Function]}
        >
          Cancel
        </Blueprint.Button>
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        >
          Save
        </Blueprint.Button>
      </div>
    </div>
  </Blueprint.Dialog>
</div>
`;
